compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingAttrs.cpp

```diff
@@ -99,6 +311,33 @@ EncodingAttr::verify(function_ref<mlir::InFlightDiagnostic()> emitError,
       }
     }
   }
+  if (iterationSizesAttr) {
+    if (!userIndexingMapsAttr) {
+      return emitError()
+             << "found `iteration_sizes` without no `user_indexing_maps`";
+    }
+    unsigned index = operandIndexAttr.getValue().getZExtValue();
+    AffineMap indexingMap;
+    if (auto map = llvm::dyn_cast<AffineMapAttr>(userIndexingMapsAttr[index])) {
+      indexingMap = map.getAffineMap();
+    } else if (auto map =
+                   llvm::dyn_cast<ArrayAttr>(userIndexingMapsAttr[index])) {
+      // This cast should be guaranteed by now based on the previous checks on
+      // `userIndexingMapsAttr`.
+      indexingMap = llvm::cast<AffineMapAttr>(map[0]).getAffineMap();
+    } else {
+      return emitError()
+             << "unexpected attribute type in `user_indexing_maps` at index: "
+             << index;
+    }
```

Comment:
I think we can declare a `SmallVector<AffineMap> indexingMaps` and store the composed map in the container. This can happen during the userIndexingMapsAttr verification (i.e., the above check). Then this snippet can be much simpler, which is just `indexingMaps[index]`.

Or we can just have a `AffineMap indexingMap` and capture the value for the `operand_index`-th composed map in the user indexing map verification.

Reply:
Yes, that's better, updated it.